bin/all_sky_search/pycbc_fit_sngls_over_multiparam

```diff
@@ -15,8 +15,64 @@
 from __future__ import division
 
 import sys, h5py, argparse, logging, pycbc.version, numpy
+from scipy.stats import norm
 from pycbc.events import triggers
 
+def dist(i1, i2, parvals, smoothing_width):
+    dsq = 0
+    for v, s in zip(parvals, smoothing_width):
+        dsq += (v[i2] - v[i1]) ** 2.0 / s ** 2.0
+    return dsq ** 0.5
+
+def smooth_templates(nabove, invalphan, ntotal, template_idx,
+                      weights=None):
+    if weights is None: weights = numpy.ones_like(template_idx)
+    nabove_t_smoothed = numpy.average(nabove[template_idx], weights=weights)
+    ntotal_t_smoothed = numpy.average(ntotal[template_idx], weights=weights)
+    invalphan_mean = numpy.average(invalphan[template_idx], weights=weights)
+
+    # return nabove_smoothed, alpha_smoothed, ntotal_smoothed
+    return_tuple = (nabove_t_smoothed,
+                    nabove_t_smoothed / invalphan_mean,
+                    ntotal_t_smoothed)
+    return return_tuple
+
+def smooth_area(nabove, invalphan, ntotal, dists):
+    idx_within_area = numpy.flatnonzero(dists < 1.)
+    return smooth_templates(nabove,
+                            invalphan,
+                            ntotal,
+                            idx_within_area)
+
+def smooth_n_closest(nabove, invalphan, ntotal, dists):
+    dist_sort = numpy.argsort(dists)
+    templates_required = 0
+    n_triggers = 0
+    # Count number of templates required to gather 1000 templates,
+    # start at closest
+    while n_triggers < 500:
```

Comment:
This should probably be un-hardcoded at some point (maybe using a smoothing-keywords option or something?), but I think 500 is probably around where it should be anyway

Reply:
yes, simplest way to enable unhardcoding would be to have eg `total_trig=500` in the function definition, I would prefer that ..